aws/resource_aws_servicecatalog_portfolio.go

```diff
@@ -0,0 +1,145 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/servicecatalog"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsServiceCatalogPortfolio() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsServiceCatalogPortfolioCreate,
+		Read:   resourceAwsServiceCatalogPortfolioRead,
+		Update: resourceAwsServiceCatalogPortfolioUpdate,
+		Delete: resourceAwsServiceCatalogPortfolioDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: false,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: false,
+			},
+			"provider_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: false,
+			},
+		},
+	}
+}
+
+func resourceAwsServiceCatalogPortfolioCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).scconn
+	input := servicecatalog.CreatePortfolioInput{}
+	if v, ok := d.GetOk("name"); ok {
+		now := time.Now()
+		input.IdempotencyToken = aws.String(fmt.Sprintf("%d", now.UnixNano()))
+		input.DisplayName = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("description"); ok {
+		input.Description = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("provider_name"); ok {
+		input.ProviderName = aws.String(v.(string))
+	}
+
+	log.Printf("[DEBUG] Creating Service Catalog Portfolio: %#v", input)
+	resp, err := conn.CreatePortfolio(&input)
+	if err != nil {
+		return fmt.Errorf("Creating Service Catalog Portfolio failed: %s", err.Error())
+	}
+	d.SetId(*resp.PortfolioDetail.Id)
+
+	return resourceAwsServiceCatalogPortfolioRead(d, meta)
+}
+
+func resourceAwsServiceCatalogPortfolioRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).scconn
+	input := servicecatalog.DescribePortfolioInput{}
+	input.Id = aws.String(d.Id())
+
+	log.Printf("[DEBUG] Reading Service Catalog Portfolio: %#v", input)
+	resp, err := conn.DescribePortfolio(&input)
+	if err != nil {
+		if scErr, ok := err.(awserr.Error); ok && scErr.Code() == "ResourceNotFoundException" {
+			log.Printf("[WARN] Service Catalog Portfolio %q not found, removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Reading ServiceCatalog Portfolio '%s' failed: %s", *input.Id, err.Error())
+	}
+	portfolioDetail := resp.PortfolioDetail
+
+	d.Set("description", portfolioDetail.Description)
```

Comment:
We should also have those attributes: `arn` (computed), `created_time` (computed)

Created time should be set as in:

```go
	if err := d.Set("created_time", portfolioDetail.CreatedTime.Format(time.RFC3339)); err != nil {
		log.Printf("[DEBUG] Error setting created_time: %s", err)
	}
```